src/peft/peft_model.py

```diff
@@ -337,6 +338,32 @@ def from_pretrained(
         else:
             raise ValueError(f"The input config must be a PeftConfig, got {config.__class__}")
 
+        weight_map = dict(named_module_tensors(model, recurse=True))
+
+        disk_modules = set()
+        index = None
+        for name, module in model.named_modules():
+            if hasattr(module, "_hf_hook") and hasattr(module._hf_hook, "original_devices"):
+                index = module._hf_hook.weights_map.dataset.index
```

Comment:
Same question as below about whether we can assume that the `index` attribute always exists.

Reply:
AFAIK every deployed `dataset` should have an `index`, but perhaps we should check for the `dataset`. I have modified for that check now